packages/mui-base/src/utils/useAnchorPositioning.ts

```diff
@@ -197,7 +199,7 @@ export function useAnchorPositioning(
     placement,
     middleware,
     strategy: positionStrategy,
-    whileElementsMounted: keepMounted ? undefined : autoUpdate,
+    whileElementsMounted: keepMounted || !trackAnchor ? undefined : autoUpdate,
```

Comment:
If for some reason they need granularity over the tracking options, we can make `trackAnchor` an object later on, and pass certain `autoUpdate` options.